test/blackbox-tests/test-cases/include-qualified/data-only-dirs-malformed.t/dune

```diff
@@ -0,0 +1,3 @@
+(include_subdirs qualified)
+(executable (name foo))
+(data_only_dirs baz-baz)
```

Comment:
What is this testing? I don't think this has any effect since there are no dune rules under `baz-baz`. (`data_only_dirs` ignores dune rules in the specified directories).

There are also no modules under that directory.

Reply:
I do think that if the directories contain no Dune rules and no modules, it would make sense for Dune to simply ignore them, making the tests in this PR irrelevant. At the time of writing this PR, though, a project of mine could not compile because of a situation like this, where there were no modules and no Dune rules yet Dune failed because of a directory with invalid name. I haven't tested on newer Dune version but I have just updated my branch and we'll see what the CI thinks of this.